homeassistant/components/http/forwarded.py

```diff
@@ -0,0 +1,176 @@
+"""Middleware to handle forwarded data by a reverse proxy."""
+from ipaddress import ip_address
+import logging
+
+from aiohttp.hdrs import X_FORWARDED_FOR, X_FORWARDED_HOST, X_FORWARDED_PROTO
+from aiohttp.web import HTTPBadRequest, middleware
+
+from homeassistant.core import callback
+
+_LOGGER = logging.getLogger(__name__)
+
+# mypy: allow-untyped-defs
+
+
+@callback
+def setup_forwarded(app, trusted_proxies):
```

Comment:
Should we name it `async_`? Our http api doesn't seem consistent when it comes to this naming pattern.
```suggestion
def async_setup_forwarded(app, trusted_proxies):
```